python/pyspark/sql/streaming/query.py

```diff
@@ -300,6 +467,24 @@ def addListener(self, listener: StreamingQueryListener) -> None:
         :class:`~pyspark.sql.streaming.StreamingQuery`.
 
         .. versionadded:: 3.4.0
+
+        Examples
+        --------
+        >>> from pyspark.sql.streaming import StreamingQueryListener
+        >>> class TestListener(StreamingQueryListener):
+        ...     def onQueryStarted(self, event):
+        ...         pass
+        ...
+        ...     def onQueryProgress(self, event):
+        ...         pass
+        ...
+        ...     def onQueryTerminated(self, event):
+        ...         pass
+
+        >>> test_listener = TestListener()
+
+        Register streaming query listener
+        >>> spark.streams.addListener(test_listener)
```

Comment:
Let's remove the listener back here too to avoid affecting other tests.

Reply:
Thanks, updated :)